repository_service_tuf/cli/admin/sign.py

```diff
@@ -105,27 +98,33 @@ def sign(
 
     1) utilizing access to the RSTUF API and signing pending metadata roles
 
-    2) provide a local file using the SIGNING_JSON_INPUT_FILE argument
+    2) provide a local file using the 'SIGNING_JSON_INPUT_FILE' argument
+
+    The result of this command will be saved locally in a 'sign-payload.json'
+    file if '--api-server' is not provided or at custom path if '--save' is
+    used.
 
     When using method 2:
 
     - 'SIGNING_JSON_INPUT_FILE' must be a file containing the JSON response
     from the 'GET /api/v1/metadata/sign' API endpoint.
 
-    - '--api_server' will be ignored.
-
-    - the result of the command will be saved into the 'sign-payload.json' file
-    unless a different name is provided with '--save'.
+    - '--api-server' will be ignored.
     """
     console.print("\n", Markdown("# Metadata Signing Tool"))
+    settings = context.obj["settings"]
+    if api_server:
+        settings.SERVER = api_server
+    if settings.get("SERVER") is None and signing_json_input_file is None:
+        error = "Either '--api-sever' or 'SIGNING_JSON_INPUT_FILE' must be set"
+        raise click.ClickException(error)
     ###########################################################################
     # Load roots
-    settings = context.obj["settings"]
-    signing_input: Optional[Dict[str, Any]] = None
+    pending_roles: Dict[str, Dict[str, Any]]
     if signing_json_input_file:
-        signing_input = json.load(signing_json_input_file)  # type: ignore
-
-    pending_roles = _get_pending_roles(settings, api_server, signing_input)
+        pending_roles = _parse_pending_data(json.load(signing_json_input_file))  # type: ignore  # noqa
+    else:
+        pending_roles = _get_pending_roles(settings)
```

Comment:
programming nit: if `_get_pending_roles` returns the parsed json, then you could call `_parse_pending_data` only once for file and api response.

Reply:
Yea, I see why you suggest that.
Unfortunately, as there are many tests already written to support the current flow it will require changes in multiple tests just to unify it to one call of `_parse_pending_data`.
For now, I will keep it like that as it's truly a nit and I have limited time for this refactoring.